### ISSUE TITLE
Question / Readme request

### ISSUE THREAD

**Issue by pihentagy** [davidmdm/myzod#54]
Hi!

Could you summarize some key differences (how to migrate, what to watch out, what is missing in myzod, which is possible in zod) if someone tries to switch from zod to myzod?

**Comment by davidmdm** [comment]
Hi, myzod's key difference is that the api is simpler and that it is more performant.

One of the key ways that it is simpler, is that it doesn't cover every Javascript type. It is solely concerned with types that you would parse from JSON or some other data format. Myzod isn't concerned with validating that an `unkown` type is a function or some instance of your class, or a symbol, because those values should never be unknown.

Unknown values are generally found at Network boundaries, file loading, essentially when you read and parse data from outside of your application.

Myzod, also doesn't copy values unless it deems that it needs to. Which is where the big performance difference between myzod and zod comes in. Personally I do not think zod is performant enough to run on high demand servers. Myzod is.
For browsers zod is fine since clients do not need to scale in the same way, and computers are fast. 

The biggest limitation with myzod is that it is currently blocked for typescript less than 4.7
At the time of writing it, I used recursive types to express type intersections `&`. However in more recent versions of typescript they have changed the heuristics such that it no longer compiles.

Personally I do not have the time to try and make it compile, and do not write much typescript these days, so I am hoping that if myzod is going to be made to work with the latest typescript versions that effort will come from the community.

the TLDR:
zod is appropriate if:
- you are on a typescript version less than 4.7
- you need fast validation like for validating incoming requests in a server
- prefer a smaller API

Zod has a bigger community and more APIs and tertiary tools.

**Comment by davidmdm** [comment]
Updating to say that the typescript version issue has been resolved and myzod is now compatible with latest released versions of typescript. 🎉

**Comment by tonyxiao** [comment]
Love the argument regarding simplicity (after all, i really don't need to validate anything that's not a JSON anyways). Good work!

**Comment by panzerchris** [comment]
One difference I noticed between zod and myzod is that the former explicitly requires "strict: true" in your typescript compiler options. There's no mention of this in myzod documentation. Are there limitations/issues caused if you use myzod while not setting `strict: true`? 

For an existing code base, setting strict: true can introduce a lot of code to fix if the code base didn't start off with this. In an ideal world, maybe all projects would, but I could see it being a helpful differentiator for some folks.

**Comment by davidmdm** [comment]
Yes that is a good callout. The same holds true for myzod. I will need to spend time in the readme soon.
